src/main/java/net/md_5/specialsource/JarRemapper.java

```diff
@@ -209,8 +209,12 @@ public void remapJar(Jar jar, File target) throws IOException {
                         buffer.flush();
                         data = buffer.toByteArray();
                     }
-                    out.putNextEntry(entry);
-                    out.write(data);
+                    try {
```

Comment:
I mean I guess its fine to include this, but it doesn't change functionality at all (see javadoc for .close and .putNextEntry - the entry will *always* be closed with the current code).

Reply:
`putNextEntry` does `closeEntry` for you, but `close` doesn't. It's quite odd.

This means that the sanity checking [here](http://grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/util/zip/ZipOutputStream.java#258) never runs for the last entry if you don't call closeEntry.

Reply:
Scratch that, I'm wrong. Sorry. Will revert this change.

`close()` calls `super.close()` which calls `finish()` which is overridden and calls `closeEntry()`